packages/vuetify/src/composables/validation.ts

```diff
@@ -0,0 +1,80 @@
+// Utilities
+import { propsFactory, wrapInPromise } from '@/util'
+import { ref, unref, watch } from 'vue'
+
+// Types
+import type { MaybeRef } from '@/util/helpers'
+import type { PropType } from 'vue'
+
+export type ValidationRule = string | ((value: any) => string | boolean)
+export type ValidationRules = ValidationRule[]
+export type ValidationResult = string | boolean | Promise<string | boolean>
+
+export interface ValidationProps {
+  errorCount: string | number
+  rules: ValidationRules
+}
+
+export const makeValidationProps = propsFactory({
+  errorCount: {
+    type: [Number, String],
+    default: 1,
+  },
+  rules: {
+    type: Array as PropType<ValidationRules>,
+    default: () => ([]),
+  },
+})
+
+export function useValidation (props: ValidationProps, value: MaybeRef<any>) {
+  const errorMessages = ref<string[]>([])
+  const isPristine = ref(true)
```

Comment:
Is pristine related to validation? I see that it flips on `isValid`, but shouldn't it flip as soon as input value changes, regardless of if it's valid or not? Or perhaps I'm misunderstanding what it represents.

Reply:
I added it so that users can determine if an input has been validated at least once. We did something similar in v2 with focusing and validation (I believe). Not committed to it.